From 31a2a33ca061e6b2188165ba2ae02752e3c8fb16 Mon Sep 17 00:00:00 2001
From: Brett Neumeier <brett@neumeier.us>
Date: Sun, 20 Dec 2009 19:23:31 -0600
Subject: [PATCH] Workaround undefined __stack_chk_guard references

When building the sysroot eglibc, the nscd Makefile tries to link
with -fstack-protector, which doesn't work.  Remove the call from
nscd-cflags.

In 2.15, this appears in more places.
---
 libc/login/Makefile  |    3 ---
 libc/nscd/Makefile   |    3 ---
 libc/resolv/Makefile |    3 ---
 3 files changed, 0 insertions(+), 9 deletions(-)

diff --git a/libc/login/Makefile b/libc/login/Makefile
index 705a451..002d471 100644
--- a/libc/login/Makefile
+++ b/libc/login/Makefile
@@ -62,9 +62,6 @@ endif
 ifeq (yesyes,$(have-fpie)$(build-shared))
 pt_chown-cflags += $(pie-ccflag)
 endif
-ifeq (yes,$(have-ssp))
-pt_chown-cflags += -fstack-protector
-endif
 ifeq (yes,$(have-libcap))
 libcap = -lcap
 endif
diff --git a/libc/nscd/Makefile b/libc/nscd/Makefile
index b9c7339..cc1416d 100644
--- a/libc/nscd/Makefile
+++ b/libc/nscd/Makefile
@@ -99,9 +99,6 @@ nscd-cflags = -DIS_IN_nscd=1 -D_FORTIFY_SOURCE=2
 ifeq (yesyes,$(have-fpie)$(build-shared))
 nscd-cflags += $(pie-ccflag)
 endif
-ifeq (yes,$(have-ssp))
-nscd-cflags += -fstack-protector
-endif
 
 CFLAGS-nscd.c += $(nscd-cflags)
 CFLAGS-connections.c += $(nscd-cflags)
diff --git a/libc/resolv/Makefile b/libc/resolv/Makefile
index 95654a4..3da9085 100644
--- a/libc/resolv/Makefile
+++ b/libc/resolv/Makefile
@@ -80,9 +80,6 @@ CPPFLAGS += -Dgethostbyname=res_gethostbyname \
 	    -Dgetnetbyname=res_getnetbyname \
 	    -Dgetnetbyaddr=res_getnetbyaddr
 
-ifeq (yes,$(have-ssp))
-CFLAGS-libresolv += -fstack-protector
-endif
 CFLAGS-res_hconf.c = -fexceptions
 
 # The BIND code elicits some harmless warnings.
-- 
1.7.3.2

